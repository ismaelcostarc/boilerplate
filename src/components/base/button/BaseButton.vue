<script setup lang="ts">
  const emit = defineEmits(['click'])
  const props = defineProps<{
    type?: string
  }>()

  const tailwindClasses = `
    ${
      props.type === 'bg-white'
        ? 'border-2 border-solid border-black'
        : 'text-white'
    }
    ${props.type || 'bg-black'}
  `

  const updateValue = () => emit('click')
</script>

<template>
  <div>
    <button
      :class="tailwindClasses"
      class="h-full w-full min-h-[2rem] min-w-[2rem] px-2 rounded-md hover:opacity-80"
      @click="updateValue"
    >
      <slot></slot>
    </button>
  </div>
</template>
