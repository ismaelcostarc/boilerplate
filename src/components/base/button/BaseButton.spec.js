import { mount } from '@vue/test-utils'
import BaseButton from './BaseButton.vue'
import { describe, it, expect, beforeEach } from 'vitest'

describe('BaseButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(BaseButton, {
      props: {
        type: 'bg-white',
      },
      slots: {
        default: 'Content',
      },
    })
  })

  it('should mount component', () => {
    expect(BaseButton).toBeTruthy()

    expect(wrapper.text()).toContain('Content')
  })

  it('should be clicked', async () => {
    expect(BaseButton).toBeTruthy()

    await wrapper.get('button').trigger('click')

    expect(wrapper.emitted()).toHaveProperty('click')
  })

  it('should be rendered correctly with props', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })

  it('should be rendered correctly without props or slots', () => {
    const wrapper1 = mount(BaseButton, {
      slots: {
        default: 'Content',
      },
    })

    const wrapper2 = mount(BaseButton, {
      props: {
        type: 'bg-white',
      },
    })

    const wrapper3 = mount(BaseButton)

    expect(wrapper1.text()).toMatchSnapshot()
    expect(wrapper2.text()).toMatchSnapshot()
    expect(wrapper3.text()).toMatchSnapshot()
  })
})
