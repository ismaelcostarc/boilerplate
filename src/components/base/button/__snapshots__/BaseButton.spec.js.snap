// Vitest Snapshot v1

exports[`BaseButton > should be rendered correctly with props 1`] = `"Content"`;

exports[`BaseButton > should be rendered correctly without props and slots 1`] = `"Content"`;

exports[`BaseButton > should be rendered correctly without props and slots 2`] = `""`;

exports[`BaseButton > should be rendered correctly without props and slots 3`] = `""`;

exports[`BaseButton > should be rendered correctly without props or slots 1`] = `"Content"`;

exports[`BaseButton > should be rendered correctly without props or slots 2`] = `""`;

exports[`BaseButton > should be rendered correctly without props or slots 3`] = `""`;
